Challo/Challo/Persistence/Repositories/Guide/Entities/GuidePersistenceObject.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  GuidePersistenceObject.swift
+//  Challo
+//
+//  Created by Kester Ng on 12/4/21.
+//
+
+import CoreData
+import Foundation
+
+struct GuidePersistenceObject {
+    let userId: UUID
+    let email: String
+    let profileImg: String?
+    let name: String?
+    let phone: String?
+    let dateJoined: Date?
+    var location: AreaPersistenceObject?
+    let sex: Sex?
+    let rating: Double
+    let daysAvailable: [Days]
+    let trails: [TrailPersistenceObject]
+    var unavailableDates: [Date]?
+    let yearsOfExperience: Int?
+    let languages: [String]?
+    let accreditations: [String]?
+    let biography: String?
+    let hobbies: String?
+    let memorableExperiences: String?
+}
```

Comment:
Is it possible to do some form of inheritance for all persistence objects? Cos there's quite abit of duplicated code for these structs

Reply:
i can do it for users i guess, cos not all are the same

Reply:
But they are. structs tho and rightfully should be structs since. they are just value types used to store in core data